### PR TITLE
MAGENTO-2296 Added route redirect logic

### PR DESCRIPTION
**Pull request by augsteyer**
## Description

Now we can pass params to the getCheckout URL like so: http://website.com/shopgate-checkout/quote/auth/token/XXX/?module=sales&controller=order&action=history, which will redirect to sales/order/history page after the token is validated.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

File: src/app/code/community/Shopgate/Cloudapi/controllers/QuoteController.php [hunk 1]
```diff
@@ -63,17 +75,19 @@ public function authAction()
          */
         $params = $this->getRequest()->getParams();
         unset($params[self::REQUEST_TOKEN_KEY]);
+        $redirect = $this->getRedirect();
+
```
Comment (Skistims) [inline src/app/code/community/Shopgate/Cloudapi/controllers/QuoteController.php]: Why did you add this extra line?

Comment (augsteyer) [reply src/app/code/community/Shopgate/Cloudapi/controllers/QuoteController.php]: no reason :D

File: src/app/code/community/Shopgate/Cloudapi/controllers/QuoteController.php [hunk 2]
```diff
@@ -87,4 +101,23 @@ protected function getCheckoutHelper()
...
+        $req    = $this->getRequest();
+        $module = $req->getParam(self::REQUEST_REDIRECT_MODULE);
+        if ($module) {
+            $controller = $req->getParam(self::REQUEST_REDIRECT_CONTROLLER, 'index');
+            $action     = $req->getParam(self::REQUEST_REDIRECT_ACTION, 'index');
+
+            return Mage::getUrl("$module/$controller/$action", array('_secure' => true));
+        }
+
+        return false;
```
[the remark this answers is not in the document]
Comment (augsteyer) [reply src/app/code/community/Shopgate/Cloudapi/controllers/QuoteController.php]: just separating concerns, trying not to mix more logic in one method than needed. I am moving more towards cleaner functions that do one thing than cleaner calls. Currently the getRedirect just handles what it needs to. The checkout url is a fallback, but it's not a fallback for getRedirect function, it's a fallback for when there are no redirect params. Not sure if it makes sense. Shorter said, if at any point we would need code like this somewhere else, it would be easier to refactor it into a helper and make it backwards compatible here.

[the remark this answers is not in the document]
Comment (augsteyer) [reply src/app/code/community/Shopgate/Cloudapi/controllers/QuoteController.php]: acceptable or you want me to refactor it more?

[the remark this answers is not in the document]
Comment (Skistims) [reply src/app/code/community/Shopgate/Cloudapi/controllers/QuoteController.php]: I guess that's fine
